common/model/transaction.h

```diff
@@ -54,6 +54,15 @@ extern "C" {
  * Transaction data structure
  ***********************************************************************************************************/
 typedef struct _iota_transaction *iota_transaction_t;
+typedef struct _iota_transaction_meta_view_model
+    *iota_transaction_meta_view_model_t;
```

Comment:
(I'm being super picky here)

Can we just do `typedef struct iota_transaction_meta_view_model_s iota_transaction_meta_view_model_t` and not involve pointer ? 

Having a typedefed pointer prevents us from being able to apply constness on the pointed data e.g.:

With typedef:
`void fn(iota_transaction_meta_view_model_t const vm)` (const pointer)

Without typedef:
`void fn(iota_transaction_meta_view_model_t const * const vm)` (const data const pointer)
or
`void fn(iota_transaction_meta_view_model_t * const vm)` (const pointer)

I know `typedef struct _iota_transaction *iota_transaction_t;` was already like that but I've been wanting to fix that for a long time :)

Reply:
https://github.com/iotaledger/entangled/issues/592

Reply:
of course